### PR TITLE
[GOBBLIn-1595]Fix the dead lock during hive registration

### PR DESCRIPTION
**Pull request by ZihanLi58**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [ ] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-1595

### Description
- [ ] Here are some details about my PR, including screenshots (if applicable):
In hive registration, we get one client and call ensureHiveTableExistenceBeforeAlternation to check table existence, and inside the method, we call tableExist method which again will try to get one client and cause dead lock. 

### Tests
- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
unit tests

### Commits
- [ ] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: gobblin-hive-registration/src/main/java/org/apache/gobblin/hive/metastore/HiveMetaStoreBasedRegister.java [hunk 1]
```diff
@@ -472,38 +472,33 @@ private boolean createTableIfNotExists(IMetaStoreClient client, Table table, Hiv
...
       }
+      this.tableAndDbExistenceCache.put(dbName + ":" + tableName, exists);
+      return exists;
     } catch (TException e) {
       throw new IOException(String.format("Unable to check existence of table %s in db %s", tableName, dbName), e);
     }
   }
 
+  @Override
+  public boolean existsTable(String dbName, String tableName) throws IOException {
```
Comment (Will-Lo) [inline gobblin-hive-registration/src/main/java/org/apache/gobblin/hive/metastore/HiveMetaStoreBasedRegister.java:496]: Small nit, can we name this function `tableExists` ? Reads a bit better

**Review by Will-Lo — COMMENTED** [review]
Small nit, otherwise LTGM

File: gobblin-hive-registration/src/main/java/org/apache/gobblin/hive/metastore/HiveMetaStoreBasedRegister.java (hunk 1 above)
Comment (ZihanLi58) [reply gobblin-hive-registration/src/main/java/org/apache/gobblin/hive/metastore/HiveMetaStoreBasedRegister.java:496]: Yeah I agree with this, but this is the API defined in HiveRegister and has been there for a long time. Change the name may make it not backward compatible and cause may downstream changes there

**Review by umustafi — APPROVED** [review]
nice job with the deadlock fix Zihan

**Pull request merged**